slang-antlr/src/test/java/com/sonarsource/slang/antlr/SLangConverterTest.java

```diff
@@ -75,20 +77,67 @@ public void simple_unary_expression() {
 
   @Test
   public void conditional_and_with_multiple_operands() {
-    BinaryExpressionTree binary = parseBinary("x && y && z");
+    BinaryExpressionTree binary = parseBinary("x && y && z;");
     assertTree(binary).isBinaryExpression(Operator.CONDITIONAL_AND);
     assertTree(binary.leftOperand()).isIdentifier("x");
     assertTree(binary.rightOperand()).isBinaryExpression(Operator.CONDITIONAL_AND);
   }
 
   @Test
   public void additive_expression_with_multiple_operands() {
-    BinaryExpressionTree binary = parseBinary("x + y\n- z");
+    BinaryExpressionTree binary = parseBinary("x + y\n- z;");
     assertTree(binary).isBinaryExpression(Operator.PLUS);
     assertTree(binary.leftOperand()).isIdentifier("x");
     assertTree(binary.rightOperand()).isBinaryExpression(Operator.MINUS).hasTextRange(1, 4, 2, 3);
   }
 
+  @Test
+  public void variable_declaration() {
+    Tree tree = converter.parse("int var x;").children().get(0);
+    Tree valueTree = converter.parse("int val x;").children().get(0);
+    Tree anotherTree = converter.parse("int var x;").children().get(0);
+    Tree yetAnotherTree = converter.parse("boolean var x;").children().get(0);
+    assertThat(tree).isInstanceOf(VariableDeclarationTree.class);
+    assertThat(anotherTree).isInstanceOf(VariableDeclarationTree.class);
+
+    VariableDeclarationTree varDeclX = (VariableDeclarationTree) tree;
+    VariableDeclarationTree anotherVarDeclX = (VariableDeclarationTree) anotherTree;
+
+    assertThat(varDeclX.children()).hasSize(2);
+    assertTree(varDeclX.type()).isIdentifier("int");
+    assertTree(varDeclX.identifier()).isIdentifier("x");
+    assertTree(varDeclX).isEquivalentTo(anotherVarDeclX);
+    assertTree(varDeclX).isNotEquivalentTo(yetAnotherTree);
+    assertTree(varDeclX).isNotEquivalentTo(valueTree);
+  }
+
+  @Test
+  public void variable_declaration_with_initializer() {
```

Comment:
You could add tests of declaration without type and also without both type and initializer (which is theoretically possible), at least to ensure `int var x = 0`, `var x = 0`, `int var x` and `var x` are not equivalent